components/brave_policy/brave_profile_policy_provider.cc

```diff
@@ -22,16 +22,18 @@
 namespace brave_policy {
 
 BraveProfilePolicyProvider::BraveProfilePolicyProvider() = default;
-BraveProfilePolicyProvider::~BraveProfilePolicyProvider() = default;
+BraveProfilePolicyProvider::~BraveProfilePolicyProvider() {
+  brave_origin::BraveOriginPolicyManager::GetInstance()->RemoveObserver(this);
+}
 
 void BraveProfilePolicyProvider::Init(policy::SchemaRegistry* registry) {
   // Call base class Init first
   policy::ConfigurationPolicyProvider::Init(registry);
 
-  // Trigger immediate policy loading to ensure policies are available in
-  // chrome://policy
-  // The actual policies will be conditionally added in LoadPolicies.
-  RefreshPolicies(policy::PolicyFetchReason::kBrowserStart);
+  // Register as BraveOriginPolicyManager observer.
+  // This ensures feature flags and local state are available before policy
+  // loading.
+  brave_origin::BraveOriginPolicyManager::GetInstance()->AddObserver(this);
```

Comment:
This class could also use scoped observation

Reply:
Sorry should have did that earlier. updated